app/controllers/sms_controller.rb

```diff
@@ -1,14 +1,51 @@
-class SmsController < ApplicationController
+class SmsController < BaseController
   # Twilio will call various endpoints in this controller
-  skip_before_filter :verify_authenticity_token
+
+  rescue_from 'QuestionValidationError' do |e|
+    render json: format_error(e), status: 500
+  end
 
   def received_user_response
     # Twilio calls this endpoint when a user sends a message to us
-    user_response = params[:Body]
-    from          = params[:From]
-    # TODO(Chris): Implement replying logic here
-    puts user_response
-    puts from
+    answer = params[:Body]
+    phone_number = params[:From]
+
+    respondent = Respondent.find_by_phone_number(phone_number.phony_normalized)
+    state = respondent.survey_execution_states.where(
+      status: [SurveyExecutionState.statuses[:initialized], SurveyExecutionState.statuses[:in_progress]]).first
+    question = state.question
+    survey = state.survey
+
+    survey_response = SurveyResponse.find_or_create_by(survey: survey, respondent: respondent)
+    response = Response.create(
+      survey_response: survey_response, respondent: respondent, question: question, answer: answer)
+
+    if question.invalid?(:next_question)
+      logger.error('Question isn\'t valid for retrieving next question, we didn\'t build the models correctly')
+      raise QuestionValidationError
+    end
+
+    if response.valid?
```

Comment:
I think this block could be cleaned up a bit. I don't see a reason to split the `Response.create` and this block given that `response` is not referenced in lines 22 to 27.

This is where `next_question` and response validation are somewhat linked because what really determines validity is checking that response is in `question.response_choices.map(&:key)`. Given that `question.next_question` will be performing a similar type of check, perhaps it makes sense to encapsulate this into `question.next_question` and catch a special error type if the provided string was not a valid response choice. If we want to prioritize decoupling the two checks over speed, then we can do an inclusion check here giving `question.next_question` the precondition that the parameter is always valid.